libRALFit/test/nlls_test.f90

```diff
@@ -1174,25 +1176,30 @@ program nlls_test
      write(options%out,*) 'Solution (C): ', x(1:n)
      write(options%out,*) 'Expected: ', (/0.3199787042575630E+00, 0.2752509146444680E-01/)
      if ( .Not. ( status%status == 0 .And. oki) ) then
-        write(*,*) 'Error: FD solve: unexpected status value or wrong solution'
+        write(*,*) 'Error: [id:2] FD solve: unexpected status value or wrong solution (Status = ', status%status,')'
+        write(*,*) 'Solution (C): ', x(1:n)
+        write(*,*) 'Expected: ', (/0.3199787042575630E+00, 0.2752509146444680E-01/)
         no_errors_main = no_errors_main + 1
      end if
 
-     ! Solve with one fixed variable
+     ! Solve with one fixed variable and solution is active
      options%Fortran_Jacobian = .False.
      options%print_level = 0
      options%check_derivatives = 0
      options%maxit = 100
+     options%fd_step = 5.0e-7
      blx(:) = (/0.0, 0.0/)
-     bux(:) = (/0.0, 1.0/)
-     X(1:2) = (/1.0, 0.923/)
+     bux(:) = (/0.0, 0.9244/)
+     X(1:2) = (/1.0, 0.92/)
      options%box_gamma = 0.99999 ! discourage any linesearch
      call solve_basic(X,params,options,status,use_fd=.True.,blx=blx,bux=bux)
-     oki = abs(x(1) - 0.0) <= 1.e-6 .And. abs(x(2)-0.9244158) < 1.e-6
```

Comment:
This change is the important diff in this file: I replace the UT solution for an active point, so the FD does not interfere with the expected solution. Mostly this is to remove the headache that each LAPACK/BLAS will return slightly different point or fail the line-search due to not tailoring the fd step.